roles/awsenv/tasks/network.yml

```diff
@@ -49,15 +50,15 @@
 
   - name: Set value of new gateway id for route table
     set_fact:
-      nat_gw_id: "{{ nat_gw.nat_gateway.nat_gateway_id }}"
+      nat_gw_id: "{{ nat_gw.nat_gateway_id }}"
     register: gw_change
 
-  when: "{{ has_nat_gw.NatGateways | count < 1 }}"
+  when: has_nat_gw.result | count < 1
 
 - name: Set value of gateway id for route table
```

Comment:
L58: maybe change to -> "Override value of gateway id for route table when 'something happened'" ?

if helps when to understand when you see the output of two actions with a very similar description, and also when reading the playbook